pkg/kube/kube_nodes.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kube
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/ethersphere/node-funder/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"math/big"
+	"net/http"
+	"os"
+)
+
+const (
+	beeAddressEndpoint = "/addresses"
+	beeWalletEndpoint  = "/wallet"
+)
+
+type NamespaceNodes struct {
+	Name  string
+	Nodes []Node
+}
+type Node struct {
+	Name      string
+	NodeIp    string
+	BeeTokens BeeTokens
+}
+
+type BeeTokens struct {
+	EthAddress      string
+	ChainID         int
+	ContractAddress string
+	ETH             *big.Int
+	BZZ             *big.Int
+	GBZZ            *big.Int
+}
+
+func NewKube() (*corev1client.CoreV1Client, error) {
+	config, err := GetConfig()
+	if err != nil {
+		log.Printf("get configuration failed with error: %v", err)
+		return nil, err
+	}
+
+	// Create a Kubernetes core/v1 client.
+	coreClient, err := corev1client.NewForConfig(config)
+	if err != nil {
+		log.Printf("creating Kubernetes client failed with error: %v", err)
+		return nil, err
+	}
+	return coreClient, nil
+}
+
+func GetConfig() (*rest.Config, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		log.Printf("obtaining user's home dir: %v", err)
+		return nil, err
+	}
+	configPath := home + "/.kube/config"
+
+	kubeConfig := flag.String("kubeconfig", configPath, "kubeconfig file")
+	flag.Parse()
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeConfig)
+	if err != nil {
+		log.Printf("creating Kubernetes client config from flag with error: %v", err)
+		return nil, err
+	}
+	return config, nil
+}
+
+func GetNodeInfo(namespace string) (*NamespaceNodes, error) {
```

Comment:
`GetNodeInfo` should also get `context` and `kube` via arguments.

Reason for passing `context`:
`GetNodeInfo` is function which makes request to k8s claster, whenever we make request we should pass it context in order to be able to end request and it's goroutines.

Reason for passing `kube`:
Kube is client for making request to k8s claster. We don't need to create this client every time when we make request, instead we should make one instance of this client and reuse it for all other requests. I see that we currently only have singe request (`GetNodeInfo`) but we should always strive to write clean code.